core/src/test/java/org/elasticsearch/action/support/replication/TransportReplicationActionTests.java

```diff
@@ -678,6 +680,91 @@ protected ReplicaResult shardOperationOnReplica(Request request) {
         assertIndexShardCounter(0);
     }
 
+    /**
+     * This test ensures that replication operations adhere to the {@link IndexMetaData#WAIT_FOR_ACTIVE_SHARDS_SETTING} setting.
+     */
+    public void testChangeWaitForActiveShardsSetting() throws Exception {
+        final String indexName = "test";
+
+        // test negative wait_for_active_shards value fails
+        expectThrows(IllegalArgumentException.class, () ->
```

Comment:
this is really the job of TRA to test this? it's an index meta data thing. Whis is it here?

I think testing here should be very minimal and just check that the TRA funnel index level setting to a request object *IF* the request has it's method set to default.